JetNews/app/src/main/java/com/example/jetnews/utils/WindowSize.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetnews.utils
+
+import androidx.compose.ui.unit.Dp
+
+/**
+ * Opinionated set of viewport breakpoints
+ *     - Compact: Most phones in portrait mode
+ *     - Medium: Most foldables and tablets in portrait mode
+ *     - Expanded: Most tablets in landscape mode
+ *
+ * More info: https://material.io/archive/guidelines/layout/responsive-ui.html
+ */
+enum class WindowSize { Compact, Medium, Expanded }
+
+fun getWindowSize(width: Dp): WindowSize = when (width.value) {
+    in 0f..599f -> WindowSize.Compact
```

Comment:
Since these are float ranges, anything greater than `599` but smaller than `600` (like `599.5`) will fall through to the `Expanded` case.

It might be clearer to just do a cascading when where the edges are clear:

```kotlin
fun getWindowSize(width: Dp): WindowSize = when {
    width < 600.dp -> WindowSize.Compact
    width <= 800.dp -> WindowSize.Medium
    else -> WindowSize.Expanded
}
```

Reply:
Yeah, sounds good. improved this and added tests